pep-0690.rst

```diff
@@ -664,17 +715,32 @@ and will be linked here soon. (TODO link.)
 Rejected Ideas
 ==============
 
-Per-module opt-in
------------------
+Wrapping deferred exceptions
+----------------------------
+
+To reduce the potential for confusion, exceptions raised in the
+course of executing a lazy import could be replaced by a ``LazyImportError``
+exception (a subclass of ``ImportError``), with a ``__cause__`` set to the
+original exception. The ``LazyImportError`` would have source location metadata
+attached pointing the user to the original import statement, to ease
+debuggability of errors from lazy imports.
+
+Ensuring that all lazy import errors are raised as ``LazyImportError`` would
+mitigate the potential confusion by reducing the likelihood that they would be
+accidentally caught and mistaken for a different expected exception. However,
+in practice we have seen cases, e.g. inside tests, where failing modules raise
+``unittest.SkipTest`` exception and this would too end up being wrapped in
+``LazyImportError``, making such tests fail because true exception types are
+being magically hidden, debunking the hypothetically likely case where
+unexpected deferred exceptions are caught by mistake as expected exceptions in
+exception handlers.
+
 
-A per-module opt-in using e.g. ``from __future__ import lazy_imports`` has a
-couple of disadvantages:
+Per-module opt-in using future imports
+--------------------------------------
 
-* It is less practical to achieve robust and significant startup-time or
-  memory-use wins by piecemeal application of lazy imports. Generally it would
-  require blanket application of the ``__future__`` import to most of the
-  codebase, as well as to third-party dependencies (which may be hard or
-  impossible.)
+A per-module opt-in using future imports (i.e.
+``from __future__ import lazy_imports`` has one big disadvantage:
 
 * ``__future__`` imports are not feature flags, they are for transition to
```

Comment:
```suggestion
A per-module opt-in using future imports (i.e.
``from __future__ import lazy_imports`` does not make sense because
``__future__`` imports are not feature flags, they are for transition to
```